src/main/reservation/main/G1HRPS/ReservationManager.java

```diff
@@ -27,24 +31,37 @@ public void RemoveFromList(Reservation reservation) {
 		throw new UnsupportedOperationException();
 	}
 
-	public void SearchList(String search_text) {
-		// TODO - implement ReservationManager.SearchList
-		throw new UnsupportedOperationException();
+	public Reservation SearchList(String search_text) {
+		for (Reservation res : reservation_list_){
+			if(res.GetReservationCode().equals(search_text)){
+				return res;
+			}
+		}
+		System.out.println("cannot find reservation to this reservation Id");
```

Comment:
There's no need to print output, the app manager can print instead. same for the others